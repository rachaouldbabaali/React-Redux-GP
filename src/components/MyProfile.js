import React from 'react';
import { useSelector } from 'react-redux';
import { selectMissions } from '../redux/Missions/missionsSlice';
import styles from '../styles/MyProfile.module.css';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rocket.rockets);
  const rocketin = [].concat(...rockets);
  const missions = useSelector(selectMissions);

  const filterRocketName = () => rocketin.filter((rocket) => rocket.reserved);

  const filterMissionByName = () => missions.filter((mission) => mission.reserved);

  return (
    <div className={styles.Wrapper}>
      <div className={styles.rocketsList}>
        <h2>My Missions</h2>
        <ul className={styles.ul}>
          {filterMissionByName().map((mission) => (
            <li key={mission.mission_id} className={styles.list}>
              {mission.mission_name}
            </li>
          ))}
        </ul>
      </div>

      <div className={styles.rocketsList}>
        <h2>My Rockets</h2>
        <ul className={styles.ul}>
          {filterRocketName().map((rocket) => (
            <li key={rocket.id} className={styles.list}>
              {rocket.name}
            </li>
          ))}
        </ul>
      </div>

    </div>
  );
};

export default MyProfile;
