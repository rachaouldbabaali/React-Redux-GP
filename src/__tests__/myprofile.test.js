import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import myStore from '../Redux/store';
import MyProfile from '../components/MyProfile';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={myStore}>
          <BrowserRouter>
            <MyProfile />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
