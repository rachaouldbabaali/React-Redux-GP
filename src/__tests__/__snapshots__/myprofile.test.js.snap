// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Wrapper"
>
  <div
    className="rocketsList"
  >
    <h2>
      My Missions
    </h2>
    <ul
      className="ul"
    />
  </div>
  <div
    className="rocketsList"
  >
    <h2>
      My Rockets
    </h2>
    <ul
      className="ul"
    />
  </div>
</div>
`;
